src/api/procedures/deletePortfolio.ts

```diff
@@ -0,0 +1,71 @@
+import BigNumber from 'bignumber.js';
+
+import { NumberedPortfolio } from '~/api/entities';
+import { PolymeshError, Procedure } from '~/base';
+import { ErrorCode } from '~/types';
+import { numberToU64, stringToIdentityId } from '~/utils';
+
+export interface DeletePortfolioParams {
+  did: string;
+  id: BigNumber;
+}
+
+/**
+ * @hidden
+ */
+export async function prepareDeletePortfolio(
+  this: Procedure<DeletePortfolioParams>,
+  args: DeletePortfolioParams
+): Promise<void> {
+  const {
+    context: {
+      polymeshApi: {
+        query: { portfolio: queryPortfolio },
+        tx: { portfolio },
+      },
+    },
+    context,
+  } = this;
+
+  const { did, id } = args;
+
+  const numberedPortfolio = new NumberedPortfolio({ did, id }, context);
+  const identityId = stringToIdentityId(did, context);
+  const rawPortfolioNumber = numberToU64(id, context);
+
+  const [isOwned, rawPortfolioName, portfolioBalances] = await Promise.all([
+    numberedPortfolio.isOwned(),
+    queryPortfolio.portfolios(identityId, rawPortfolioNumber),
```

Comment:
Use the portfolio method that gets the name (rebase to your other branch if needed)